drivers/serial/uart_nrfx_uarte.c

```diff
@@ -39,39 +39,50 @@ LOG_MODULE_REGISTER(uart_nrfx_uarte, CONFIG_UART_LOG_LEVEL);
 #error "No PPI or DPPI"
 #endif
 
+/* We cannot use LISTIFY because LISTIFY cannot be used recursively and it is already
+ * used inside macros used by UARTE device instantiation so we list all existing
+ * IDs used for UARTE peripheral in all Nordic devices.
+ */
+#define ALL_UARTE_INSTANCES \
+	0, 1, 2, 3, 00, 20, 21, 22, 30, 120, 130, 131, 132, 133, 134, 135, 136, 137
+
+/* Execute macro f(x) for all instances. */
+#define UARTE_FOR_EACH_INSTANCE(f, sep) \
+	FOR_EACH(f, sep, ALL_UARTE_INSTANCES)
```

Comment:
Couldn't `NRFX_FOREACH_ENABLED(UARTE, ...)` be used for this as in `uart_nrfx_uarte2.c`? The above imposes maintaining the list of possible UARTE instances in this file.

Reply:
switched to use `NRFX_FOREACH_PRESENT` as `FOREACH_ENABLED` cannot be used because this applies to enabled nrfx driver instances.